src/sentry/static/sentry/app/views/organizationJoinRequest.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import styled from 'react-emotion';
+import {Params} from 'react-router/lib/Router';
+
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {t, tct} from 'app/locale';
+import EmailField from 'app/views/settings/components/forms/emailField';
+import Form from 'app/views/settings/components/forms/form';
+import InlineSvg from 'app/components/inlineSvg';
+import NarrowLayout from 'app/components/narrowLayout';
+import space from 'app/styles/space';
+
+type Props = {
+  params: Params;
+};
+
+type State = {
+  submitSuccess: boolean | null;
+};
+
+class OrganizationJoinRequest extends React.Component<Props, State> {
+  state: State = {
+    submitSuccess: null,
+  };
+
+  handleSubmitSuccess = () => {
+    this.setState({submitSuccess: true});
+  };
+
+  handleSubmitError() {
+    addErrorMessage(t('Request to join failed'));
+  }
+
+  handleCancel = e => {
+    e.preventDefault();
+
+    const {orgId} = this.props.params;
+    window.location.assign(`/auth/login/${orgId}/`);
+  };
+
+  render() {
+    const {orgId} = this.props.params;
+    const {submitSuccess} = this.state;
+
+    return (
+      <React.Fragment>
+        {submitSuccess ? (
```

Comment:
A nicer way to write this might be to have a conditional before this return to check submit success and return just this success modal:

```jsx
render() {
  if (submitSuccess) {
    return ( ... )
  }

  return (
    // the form
  )
}
```